frontend/src/components/DiscussionListByData.vue

```diff
@@ -0,0 +1,114 @@
+<template>
```

Comment:
This is a duplicate of DiscussionList without the data fetching. The correct way for this refactor is to make DiscussionList use DiscussionListByData. 

Also, DiscussionList is a better name for a component that only renders UI and data fetching can be done separately.